iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/extractor/dataregion/realtime/epoch/TsFileEpochManager.java

```diff
@@ -79,7 +86,24 @@ public PipeRealtimeEvent bindPipeInsertNodeTabletInsertionEvent(
     return new PipeRealtimeEvent(
         event,
         epoch,
-        Collections.singletonMap(node.getDevicePath().getFullPath(), node.getMeasurements()),
+        node instanceof InsertRowsNode
+            ? getDevice2MeasurementsMapFromInsertRowsNode((InsertRowsNode) node)
```

Comment:
Can be replaced with "insertRowNodeList.stream()
        .collect(
            Collectors.toMap(
                insertRowNode -> insertRowNode.getDevicePath().getFullPath(),
                InsertNode::getMeasurements,
                (oldMeasurements, newMeasurements) ->
                    Stream.of(Arrays.asList(oldMeasurements), Arrays.asList(newMeasurements))
                        .flatMap(Collection::stream)
                        .distinct()
                        .toArray(String[]::new)));"

Reply:
fixed